pkg/ingester/user_state.go

```diff
@@ -36,26 +37,36 @@ type userState struct {
 	userID              string
 	fpLocker            *fingerprintLocker
 	fpToSeries          *seriesMap
+	metricToMetadata    *metadataMap
```

Comment:
Design question. Is there the strong need to couple the metadata ingestion implementation within `userState`? I'm asking because the blocks-storage doesn't use `userState` (only its lock for other purposes) and my plan would be get rid completely of `userState` in the blocks-storage implementation (before this PR `userState` was strictly related to chunks storage).

Would be more difficult to implement the metadata ingestion outside of `userState`?

Reply:
Let me know what you think here, I've tried to keep everything in one place - we can worry about extracting methods for re-using after we get the coupling right.